client/src/component/sos/audit/RiskLimitForm.tsx

```diff
@@ -8,63 +10,117 @@ interface FormProps {
 }
 
 interface FormState {
+    ballotPollingField: string;
     ballotPollingLimit: number;
+    comparisonField: string;
     comparisonLimit: number;
 }
 
 
 const DEFAULT_RISK_LIMIT = 0.05;
+const MIN_RISK_LIMIT = 0.0001;
+const MAX_RISK_LIMIT = 1 - MIN_RISK_LIMIT;
+
+
+function isValidRiskLimit(limit: number) {
+    return isFinite(limit)
+        && MIN_RISK_LIMIT <= fromPercent(limit)
+        && fromPercent(limit) <= MAX_RISK_LIMIT;
+}
+
+function fromPercent(val: number) {
+    return val / 100;
+}
+
+function toPercent(val: number) {
+    return val * 100;
+}
 
 
 class RiskLimitForm extends React.Component<FormProps & any, FormState> {
     public state: FormState = {
+        ballotPollingField: `${toPercent(DEFAULT_RISK_LIMIT)}`,
         ballotPollingLimit: DEFAULT_RISK_LIMIT,
+        comparisonField: `${toPercent(DEFAULT_RISK_LIMIT)}`,
         comparisonLimit: DEFAULT_RISK_LIMIT,
     };
 
     public render() {
-        const { ballotPollingLimit, comparisonLimit } = this.state;
+        const {
+            ballotPollingField,
+            ballotPollingLimit,
+            comparisonField,
+            comparisonLimit,
+        } = this.state;
 
         this.props.forms.riskLimit = this.state;
 
         return (
             <div>
                 <div>
                     <label>
-                        Ballot Polling Audits
+                        Ballot Polling Audits (%)
                         <NumericInput
-                            minorStepSize={ 0.001 }
-                            stepSize={ 0.01 }
-                            value={ ballotPollingLimit }
+                            allowNumericCharactersOnly={ true }
+                            min={ toPercent(MIN_RISK_LIMIT) }
+                            max={ toPercent(MAX_RISK_LIMIT) }
+                            minorStepSize={ toPercent(0.001) }
```

Comment:
What are the visual implications of these step sizes?

Reply:
@kiniry these determine what happens when you use click up and down arrows (`stepSize`) or shift-click (`minorStepSize`).

Reply:
Ah!  *shift*-click!  But with a 2 digit precision, one will not witness any visual feedback from such a UI action, right?

Reply:
Correction: minor is for `alt`-click. We _do_ observe the expected step sizes (1% and 0.1% for `step` and `minorStep`), because I convert the steps to percentage values.